src/optimizer.rs

```diff
@@ -205,35 +204,6 @@ fn decat(n: &mut Node, _w: &Walk) -> PassAction {
     }
 }
 
-/// Unfold icase chars.
-/// That means for case-insensitive characters, figure out everything that they
-/// could match.
-/// TODO: should cache unfolding.
-fn unfold_icase_chars(n: &mut Node, _w: &Walk) -> PassAction {
```

Comment:
This is (was) a big optimization - this replaces a case-insensitive character match with with a set of all characters that match in a case-insensitive way. This changes pushes the computation to matching time.

The optimization is important, because many characters only match themselves when case-insensitive.

Is there no way to preserve this optimization - to have icu4x find all characters that match a given character, ignoring case?